src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java

```diff
@@ -237,6 +238,23 @@ public boolean hasResident(final CitizenData citizen)
     }
 
     /**
+     * Checks if food in the home is required.
+     * If yes set foodNeeded to true, else to false.
+     */
+    public void checkIfFoodNeeded()
+    {
+        boolean needFood = false;
+        for(final CitizenData resident: residents)
```

Comment:
This can simply be done by a lamda (with findFirst and filter)
Would make this Code footprint a lot smaller and the JVM can optimize it better.